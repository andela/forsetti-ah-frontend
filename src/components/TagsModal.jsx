import React, { Component } from 'react';
import { connect } from 'react-redux';
import TagsInput from 'react-tagsinput';
import {
  Button,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter
} from 'reactstrap';
import { closeTagsModal } from '../actions';

class TagsModal extends Component {
    state = { tags: [] };

    handleChange = (tags) => {
      this.setState({ tags });
    }

    closeModal = () => {
      const { closetagsmodal } = this.props;
      closetagsmodal();
    }

    handleSubmit = () => {
      const { publishHandler } = this.props;
      const { tags } = this.state;
      this.closeModal();
      publishHandler(tags);
    }

    render() {
      const { modal, tags } = this.state;
      const { className, tagsmodal } = this.props;
      return (
        <div>
          <Modal isOpen={tagsmodal.showTagsModal} toggle={this.closeModal} className={className}>
            <ModalHeader toggle={this.closeModal}>Add Tags(Optional)</ModalHeader>
            <ModalBody>
              <p>You can separate your tags with either of the SPACEBAR or ENTER keys.</p>
              <TagsInput
                value={tags}
                onChange={this.handleChange}
                addKeys={[13, 32]}
                maxTags={10}
              />
            </ModalBody>
            <ModalFooter>
              <Button color='info' onClick={this.handleSubmit}>Submit</Button>
            </ModalFooter>
          </Modal>
        </div>
      );
    }
}
const mapStateToProps = state => ({
  tagsmodal: state.modal
});

const mapDispatchToProps = {
  closetagsmodal: closeTagsModal
};

export default connect(mapStateToProps, mapDispatchToProps)(TagsModal);
