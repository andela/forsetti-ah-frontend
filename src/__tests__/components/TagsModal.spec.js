import React from 'react';
import { shallow, mount } from 'enzyme';
import TagsModal from '../../components/TagsModal';

describe('Tags Modal', () => {
  let wrapper;
  const props = {
    tagsmodal: { showTagsModal: false },
    closetagsmodal: jest.fn(),
    publishHandler: jest.fn()
  };

  test('Should render component successfully', () => {
    wrapper = shallow(<TagsModal.WrappedComponent {...props} />);
    expect(wrapper.length).toEqual(1);
  });

  test('Should maintain existing snapshot', () => {
    wrapper = shallow(<TagsModal.WrappedComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should correctly set state for tags modal', () => {
    wrapper = shallow(<TagsModal.WrappedComponent {...props} />);

    const tagsInput = wrapper.find('TagsInput');
    tagsInput.simulate('change', ['tag1', 'tag2']);

    expect(wrapper.state().tags).toEqual(['tag1', 'tag2']);
  });

  test('should call appropriate function on submit', () => {
    wrapper = shallow(<TagsModal.WrappedComponent {...props} />);

    const submitButton = wrapper.find('Button');
    submitButton.simulate('click');

    expect(props.publishHandler).toHaveBeenCalled();
  });
});
