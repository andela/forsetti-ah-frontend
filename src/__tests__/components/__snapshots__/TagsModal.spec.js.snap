// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags Modal Should maintain existing snapshot 1`] = `ShallowWrapper {}`;
